CMakeLists.txt

```diff
@@ -90,7 +90,8 @@ OPTION(XNNPACK_ENABLE_RISCV_VECTOR "Build XNNPACK with RISC-V Vector micro-kerne
 OPTION(XNNPACK_ENABLE_VSX "Build XNNPACK with VSX Vector micro-kernels for Power" ON)
 SET(XNNPACK_AVXVNNI_DEFAULT ON CACHE INTERNAL "Default value to enable AVX-VNNI in x86 builds.")
 SET(XNNPACK_AVX512AMX_DEFAULT ON CACHE INTERNAL "Default value to enable AVX512-AMX in x86 builds.")
-IF(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+# Note that support for AMX was added in version 11 of Clang and version 11 of GCC.
+IF(CMAKE_C_COMPILER_ID STREQUAL "GNU" OR CMAKE_C_COMPILER_ID STREQUAL "Clang")
```

Comment:
the XNNPACK_AVXVNNI_DEFAULT variable doesnt seem to apply to the option.  I've refactored this to set the option first and then update the option if the compiler is old./
Also the more common issue is the amxintrinsic.h only defines intrinsics for x86_64 regardless of OS